src/midi/midi_handler.h

```diff
@@ -47,6 +48,24 @@ class MidiHandler {
 
 	virtual void Close() {}
 
+	void HaltSequence()
+	{
+		uint8_t message[MIDI_MSG_SIZE] = {};
```

Comment:
Do we really need MIDI_MSG_SIZE here? We use only 2 bytes.

Reply:
Unlike PlaySyse(*, len), the PlayMsg(*) API takes in a buffer (pointer) but doesn't let you specify the buffer length. 

The current MIDI structure uses a fixed message buffer of eight characters, and that's proven to work cross-platform with all kinds of different MIDI players. So that's what I'm using here too.

But you're right: the MIDI spec says messages can be two or three byte sequences, and they don't even mention providing a terminatinf character (such as 0x0).  So I could try a strict 2-byte array, and if the MIDI software tries reading the 3rd byte (it will bus error?).

Reply:
Yeah, I don't understand why there are 8 bytes in there…

I am perfectly OK with having 3 bytes ending with 0-byte :) I think all our backends only look up up to 3 bytes at a time.

Reply:
More digging into how the MIDI device knows out when to stop reading this buffer.

Turns out the first byte is looked-up in a length table:

```

uint8_t MIDI_evt_len[256] = {
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x00
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x10
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x20
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x30
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x40
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x50
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x60
  0,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,0,  // 0x70

  3,3,3,3, 3,3,3,3, 3,3,3,3, 3,3,3,3,  // 0x80
  3,3,3,3, 3,3,3,3, 3,3,3,3, 3,3,3,3,  // 0x90
  3,3,3,3, 3,3,3,3, 3,3,3,3, 3,3,3,3,  // 0xa0
  3,3,3,3, 3,3,3,3, 3,3,3,3, 3,3,3,3,  // 0xb0

  2,2,2,2, 2,2,2,2, 2,2,2,2, 2,2,2,2,  // 0xc0
  2,2,2,2, 2,2,2,2, 2,2,2,2, 2,2,2,2,  // 0xd0

  3,3,3,3, 3,3,3,3, 3,3,3,3, 3,3,3,3,  // 0xe0

  0,2,3,2, 0,0,1,0, 1,0,1,1, 1,0,1,0   // 0xf0
};
```

Our halt sequence always starts with `0xb`, so our buffer should be three bytes. 

Also double checked that `PlayMsg(*)` implementations will indeed expect to read `message[2]` (in this case), but no further.

``` c++
void MidiHandler_oss::PlayMsg(const uint8_t *msg)
{
	const uint8_t len = MIDI_evt_len[*msg];
	uint8_t buf[128];
	assert(len * 4 <= sizeof(buf));
	size_t pos = 0;
	for (uint8_t i = 0; i < len; i++) {
		buf[pos++] = SEQ_MIDIPUTC;
		buf[pos++] = *msg;
		buf[pos++] = device_num;
		buf[pos++] = 0;
		msg++;
	}
	write(device, buf, pos);
}
```